beacon-chain/blockchain/process_block.go

```diff
@@ -529,11 +532,24 @@ func (s *Service) runLateBlockTasks() {
 	}
 }
 
+func missingIndices(bs *filesystem.BlobStorage, root [32]byte, expected int) (map[uint64]struct{}, error) {
+	indices, err := bs.Indices(root)
+	if err != nil {
+		return nil, err
+	}
+	missing := make(map[uint64]struct{}, expected)
+	for i := uint64(0); i < uint64(expected); i++ {
+		if !indices[i] {
```

Comment:
indices is a fixed size, always equal to `MAX_BLOBS_PER_BLOCK`.
expected comes from blocks, and there are validation rules that enforce the kzg_commitments field to be `< MAX_BLOBS_PER_BLOCK`, eg in gossip validation we have `[REJECT] The sidecar's index is consistent with MAX_BLOBS_PER_BLOCK -- i.e. blob_sidecar.index < MAX_BLOBS_PER_BLOCK.`.

Usually we can assume ssz encoding acts as a validity check - ie you couldn't unmarshal a Vector beyond its fixed size, or a List exceeding its max size. But this field is weird, because the ssz is defined to match the EL theoretical max size (4096) not the current CL size limit (6). So maybe we should still do a defensive check here.

Reply:
defensive check here:
https://github.com/prysmaticlabs/prysm/pull/13190/commits/f3b3afcb0f1a8d7400a2dd8b043285ae1151cc32
and I think passing in the expected commitment slice instead of the length of it might make things slightly more clear:
https://github.com/prysmaticlabs/prysm/pull/13190/commits/71e65af6839d5726bd023fab5d3b33898e385b92